core/src/main/java/io/paradoxical/cassieq/dataAccess/QueueRepositoryImpl.java

```diff
@@ -42,175 +39,146 @@ public QueueRepositoryImpl(@NonNull final Session session) {
+                                               .value(Tables.Queue.VERSION, 0)
+                                               .value(Tables.Queue.BUCKET_SIZE, queueDefinition.getBucketSize().get())
+                                               .value(Tables.Queue.QUEUE_NAME, queueDefinition.getQueueName().get())
+                                               .value(Tables.Queue.MAX_DELIVERY_COUNT, queueDefinition.getMaxDeliveryCount())
+                                               .value(Tables.Queue.STATUS, QueueStatus.Inactive.ordinal());
 
-        // if however, an inactve queue exists, make sure only one person can
-        // create the next active queue (prevent race conditions of multiple queues being created
-        // of the same name that are active
-        // first check the table name version table
-        final Select.Where activeQueueIdSelect = QueryBuilder.select()
-                                                             .column(Tables.QueueNameManager.VERSION)
-                                                             .from(Tables.QueueNameManager.TABLE_NAME)
-                                                             .where(eq(Tables.QueueNameManager.QUEUE_NAME, queueDefinition.getQueueName().get()));
+        final boolean queueInserted = session.execute(insertQueue).wasApplied();
 
-        final Row row = session.execute(activeQueueIdSelect).one();
+        if (queueInserted) {
 
-        int currentVersion;
+            resetQueueMonotonicValue(queueDefinition.getQueueName());
 
-        if (row == null) {
-            currentVersion = seedQueueNameIdGeneratorTable(queueDefinition);
-        }
-        else {
-            currentVersion = row.getInt(0);
+            resetQueuePointers(queueDefinition.getQueueName());
+
+            return trySetQueueDefinitionStatus(queueDefinition.getQueueName(), QueueStatus.Active);
         }
 
-        final int nextVersion = currentVersion + 1;
+        return false;
+    }
 
-        // update the tracking table to see who can grab the next version
-        // only if the queue name status is inactive
-        // if its available grab it
-        final Statement insertTrackingId = QueryBuilder.update(Tables.QueueNameManager.TABLE_NAME)
-                                                       .where(eq(Tables.QueueNameManager.QUEUE_NAME, queueDefinition.getQueueName().get()))
-                                                       .with(set(Tables.QueueNameManager.VERSION, nextVersion))
-                                                       .and(set(Tables.Queue.STATUS, QueueStatus.Active.ordinal()))
-                                                       .onlyIf(eq(Tables.QueueNameManager.VERSION, currentVersion))
-                                                       .and(eq(Tables.QueueNameManager.STATUS, QueueStatus.Inactive.ordinal()));
+    private boolean upsertQueueDefinition(@NonNull final QueueDefinition queueDefinition) {
```

Comment:
What do you think @devshorts? not having two tables to manage makes this way simpler. The general idea is:
1. insert the queue, if we can.
2. if insert failed, reset all the pointers/monoton. (this isn't needed if we leave the pointer/monotons to use queue ids...)
3. update state to active with new version and settings.